site/content/guides/routing.md

```diff
@@ -0,0 +1,285 @@
+
+Let's start by creating components. We will have two components for content pages: `HomePage` and
+`AboutPage`. We are also going to have `RootComponent` which will serve as 
+container for navigation links and content pages.
+
+We will need to import `React` for creating components as well as `Link` and `Route` components
+from `frint-router-react`.
+
+```js
+import React from 'react';
+import { Link, Route } from 'frint-router-react';
+
+const HomePage = () => <div>Home page</div>;
+const AboutPage = () => <div>About page</div>;
+
+const RootComponent = () => {
+  return (
+    <div>
+      <div className="links">
+        <Link to="/" activeClassName="is-active" exact>[Home]</Link>
+        <Link to="/about" activeClassName="is-active">[About]</Link>
+      </div>
+
+      <div className="content">
+        <Route path="/" component={HomePage} exact />
+        <Route path="/about" component={AboutPage} />
+      </div>
+    </div>
+  );
+};
+```
+
+Let's go over what happens here.
+
+1. `RootComponent` renders section with navigation links using `Link` component. `Link` component 
+creates an anchor or a button leading to a certain URL. It also applies CSS class `is-active` 
+(passed as `activeClassName` prop) to the link element automatically when current URL matches the 
+one passed to the `Link`.
+2. It also makes use of `Route` component to switch between content pages. Whenever `path` passed to
+`Route` component matches current URL it will render `component` from the `props`. Just as with 
+`Link` you can choose whether it should be an exact match. You can also pass a Frint app as an `app`
+ prop instead of `component`.
+
+### Creating a router
+
+To create a router you need to import one of the router services from `frint-router` package and 
+instantiate it. 
+
+There are three router services available:
+- `BrowserRouterService`: uses modern HTML5 History API
+- `HashRouterService`: for legacy browsers
+- `MemoryRouterService`: useful for tests
+
+For the purpose of this tutorial we will use `BrowserRouterService`: 
+
+```js
+import BrowserRouterService from 'frint-router/BrowserRouterService';
+
+const router = new BrowserRouterService();
+```
+
+### Creating an app
+
+Now let's create an app that makes use of the `RootComponent` and `router` we've just created. We
+pass them as providers to the app. Names of the providers are defined here by convention. 
+
+- `frint-react` requires that app's main component would be a provider named `component`.
+- `frint-router-react` relies on `router` provider name for `Link` and `Route` to get and set 
+current URL. If you want to use `Link` or `Route` in your child apps you'll need to cascade the 
+`router` provider to them by adding `cascade: true`   
+
+```js
+import { createApp } from 'frint';
+
+const RouterApp = createApp({
+  name: 'RouterApp',
+  providers: [
+    {
+      name: 'component',
+      useValue: RootComponent
+    },
+    {
+      name: 'router',
+      useValue: router,
```

Comment:
it's a good advantage, I agree. but you can also pass `options` to the new class instance in a very straightforward way when you create it to be later used in `useValue`:
```
const router = new BrowserRouterService(options);
```
and it doesn't add complexity by introducing factory. although I agree the complexity is pretty minimal, if we want to target as many devs as possible, I would avoid any